wire-schema/src/jvmMain/java/com/squareup/wire/schema/PruningRules.kt

```diff
@@ -125,28 +126,27 @@ class PruningRules private constructor(builder: Builder) {
     var excludeMatch: String? = null
     var rule: String? = identifier
     while (rule != null) {
-      if (excludes.contains(rule)) {
+      if (excludeMatch == null && excludes.contains(rule)) {
         excludeMatch = rule
       }
-      if (includes.contains(rule)) {
+      if (includeMatch == null && includes.contains(rule)) {
         includeMatch = rule
       }
       rule = enclosing(rule)
     }
 
-    return when {
-      excludeMatch != null -> {
-        usedExcludes.add(excludeMatch)
-        false
-      }
-      includeMatch != null -> {
-        usedIncludes.add(includeMatch)
-        true
-      }
-      else -> {
-        false
-      }
+    val isRoot = when {
+      excludeMatch != null && includeMatch != null -> excludeMatch.length < includeMatch.length
```

Comment:
I think it matters if one is a type and the other is not. This is relevant when pruning the field type vs. the enclosing type. In other words, `a.b.C#d` is effectively longer than `a.b.e.Elephant` even though the string is shorter.

Reply:
When would that happen though? the longest identifier will always contain the short one if we have both an `excludeMatch` and an `includeMatch`, they cannot both match without one containing the other.

Reply:
When would that happen though? the longest identifier will always contain the short one if we have both an excludeMatch and an includeMatch, they cannot both match without one containing the other.